### DISCUSSION TITLE
UI Field should be able to manage "data"

### DISCUSSION THREAD

**Discussion by finkinfridom** [payloadcms/payload#1250] [Feature Requests & Ideas]
I was trying to create a "UI" field in the users' collection to show the profile picture (if any).
Unfortunately, there's no way to define hooks or to receive the "data" payload at render time.

Custom UI fields could be beneficial for visualizing logos, pictures, and graphical elements to admin users.

**Comment by jmikrut** [comment]
Hey @finkinfridom — you actually can do this no problem.

Have a look at all the hooks available to you here:
https://payloadcms.com/docs/admin/hooks

All of these can be used within a `ui` field and they allow you to both send and receive data from the form.

**Reply by finkinfridom** [reply]
cool! 
thanks for the suggestion. 
I think my misread comes from the different "hooks" it's possible to setup (I expected to find something at field level: https://payloadcms.com/docs/hooks/fields)
